File: leetcode/easy/226-Invert-Binary-Tree.py
#!/usr/bin/python3
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    # Recursive Solution
    def invertTree(self, root: list) -> list:
      if root:
        right = root.right
        root.right = root.left
        root.left = right
        self.invertTree(root.right)
        self.invertTree(root.left)
        return root
    
        

File: neetcode150/easy/cpp/226-Invert-Binary-Tree.cpp
// Definition for a binary tree node.
#include <algorithm>

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        if (root == NULL) {
            return NULL;
        }
        std::swap(root->right, root->left);
        invertTree(root->left);
        invertTree(root->right);
        return root;
    }
};